solr/core/src/test/org/apache/solr/update/processor/SkipExistingDocumentsProcessorFactoryTest.java

```diff
@@ -314,6 +314,22 @@ public void testSkippableUpdateIsSkippedIfSkipUpdatesTrue() throws IOException {
     verify(next, never()).processAdd(cmd);
   }
 
+  @Test
```

Comment:
Thanks for adding tests.  This is more of a personal comment, but I find mocks to be harder to understand than more traditional tests (that in solr world do end up being integration sytle).   Mostly because I sometimes stugglre to understand the data set up....   i know this class has other mocks as well...

Reply:
True, these are pure unit tests, so we're not reliant on actual indexes and calls to RTG code via static methods that we can't intercept during tests.

I guess this is the result of our internal testing strategies (pure unit tests, plus docker-based full integration tests) but I'm happy to add real index testing if there's a standard base test I should build on?

Reply:
I think that if there are some good examples in `SkipExistingDocumentsProcessorFactoryTest` that aren't mock, that you can use, then great...   However, if not, and these mocks do work, then thats fine.   

Someday, I'd love to see some more comprehensive full integration tests added to Solr...    There are some BATS tests that start to approach that, but we don't have consensus on it.

Reply:
I have faith that this unit test works so it's good enough for me (don't need the expense of a higher level test).  It's easy to read too.
Presumably you've checked that the test would fail were it not for your changes?